### PR TITLE
Logging update

### PR DESCRIPTION
**Pull request by Skptak**
Swapping this to use a parent and child thread to handle the case where readline() blocks. 
Change to some logic of it.

### CODE REVIEW

**Review by A-Zaba — DISMISSED** [review]

**Comment by Skptak** [comment]
Log of these changes working for FreeRTOS+TCP Demos can be found [here](https://github.com/ChaiTowKway/FreeRTOS/actions/runs/5616989789)

File: executable-monitor/action.yml [hunk 1]
```diff
@@ -19,14 +20,59 @@ inputs:
...
+          fi
         shell: bash
-      - name: Run executable with monitoring script
+      - name: Run Executable with Monitoring
         run: |
-          python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} --success-line="${{ inputs.success-line }}" --log-dir=${{ inputs.log-dir }} --retry-attempts=${{ inputs.retry-attempts }}
+          # Run Executable with Monitoring
+          echo "::group::Executable Output"
+          echo "inputs.success-exit-code:${{ inputs.success-exit-code }}\n"
+          echo "inputs.success-line:${{ inputs.success-line }}\n"
```
Comment (AniruddhaKanhere) [inline executable-monitor/action.yml:50]: Why are we printing these things?

File: executable-monitor/action.yml [hunk 2]
```diff
@@ -19,14 +20,59 @@ inputs:
...
+            optArgs="--log-dir=${{ inputs.log-dir}} $optArgs"
+          fi
+          echo "optArgs= $optArgs"
+          if [ "${{ inputs.success-line }}" != "" ]; then
+            python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} $optArgs --success-line="${{ inputs.success-line}}"
+          else
+            python3 $GITHUB_ACTION_PATH/executable-monitor.py --exe-path=${{ inputs.exe-path }} --timeout-seconds=${{ inputs.timeout-seconds }} $optArgs
+          fi
+          echo "::endgroup::"
+          if [ "$?" = "0" ]; then
```
Comment (AniruddhaKanhere) [inline executable-monitor/action.yml:71]: Shouldn't we check against `${{ inputs.success-exit-code }}`?
Or am I not understanding this correctly?

File: executable-monitor/executable-monitor.py [hunk 3]
```diff
@@ -36,8 +149,8 @@
     parser.add_argument('--success-line',
                         type=str,
                         required=False,
-                        help='Line that indicates executable completed successfully. Required if --success-exit-status is not used.')
-    parser.add_argument('--success-exit-status',
+                        help='Line that indicates executable completed successfully. Required if --success-exit-code is not used.')
+    parser.add_argument('--success-exit-code',
                         type=int,
                         required=False,
                         help='Exit status that indicates that the executable completed successfully. Required if --success-line is not used.')
```
Comment (AniruddhaKanhere) [inline executable-monitor/executable-monitor.py:43]: Don't we need these parsers as we are passing in these parameters?

Comment (AniruddhaKanhere) [reply executable-monitor/executable-monitor.py:43]: Saw that this was added later in the file. Marking this comment as resolved.

File: executable-monitor/action.yml (hunk 2 above)
Comment (Skptak) [reply executable-monitor/action.yml:71]: The executable monitor itself does this check, where from here I want to know if the python application exited correctly

File: executable-monitor/executable-monitor.py (hunk 3 above)
Comment (Skptak) [reply executable-monitor/executable-monitor.py:43]: I changed it from "success-exit-status "to "success-exit-code" is all

File: executable-monitor/action.yml (hunk 1 above)
Comment (Skptak) [reply executable-monitor/action.yml:50]: Was in there from previous debugging, removed them

File: executable-monitor/action.yml (hunk 2 above)
Comment (AniruddhaKanhere) [reply executable-monitor/action.yml:69]: Aha, I see.

File: executable-monitor/action.yml (hunk 1 above)
Comment (AniruddhaKanhere) [reply executable-monitor/action.yml]: Thanks

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by rawalexe — APPROVED** [review]

File: executable-monitor/executable-monitor.py (hunk 3 above)
Comment (AniruddhaKanhere) [reply executable-monitor/executable-monitor.py:43]: I see my folly - meant to put this comment elsewhere :D

**Pull request merged**